lib/ansible/plugins/lookup/config.py

```diff
@@ -56,32 +71,81 @@
   type: raw
 """
 
+import ansible.plugins.loader as plugin_loader
+
 from ansible import constants as C
-from ansible.errors import AnsibleError
+from ansible.errors import AnsibleError, AnsibleLookupError, AnsibleOptionsError
+from ansible.module_utils._text import to_native
 from ansible.module_utils.six import string_types
 from ansible.plugins.lookup import LookupBase
+from ansible.utils.sentinel import Sentinel
+
+
+class MissingSetting(AnsibleError):
+    pass
+
+
+def _get_plugin_config(pname, ptype, config, variables):
+    try:
+        # plugin creates settings on load, this is cached so not too expensive to redo
+        loader = getattr(plugin_loader, '%s_loader' % ptype)
+        dump = loader.get(pname, class_only=True)
```

Comment:
Can you add tests for plugins in collections? I'm struggling to get it working, but might just be overlooking something. Does this work with doc fragments?

Reply:
Oh, I guess this is an inventory-specific oddity. Unsure if this should handle `inventory` as a plugin type since when inventory decides to load options isn't until parse time, regardless of whether the option is set via environment var...

Reply:
should not work for doc_fragments directly since they are not really plugins, but it should load them normally for other plugins as that happens in plugin_loader itself during normal execution.

Reply:
^ this will work when overlayed with the all_plugins_config PR that makes Options error 'handled'